### PR TITLE
Optimize: VM implementation "switch case" to "delegate array"

### PR DESCRIPTION
**Pull request by Akeit0**
This PR is a large optimization.
NBodyBenchmark 
.NET8  12ms -> 6.2ms
IL2CPP 81ms ->15ms

### CODE REVIEW

**Comment by Akeit0** [comment]
All changes are internal, so I think nothing is broken.

File: src/Lua/Runtime/LuaVirtualMachine.cs [hunk 1]
```diff
@@ -1,1073 +1,1659 @@
...
+                {
+                    var stack = context.Stack;
+                    var RA = context.Instruction.A + context.Frame.Base;
+                    stack.Get(RA) = context.TaskResult == 0 ? LuaValue.Nil : context.ResultsBuffer[0];
+                    stack.NotifyTop(RA + 1);
+                };
+            }
+            else
+            {
+                Console.WriteLine($"mul {instruction.B} {vb} {instruction.C} {vc}");
```
Comment (AnnulusGames) [inline src/Lua/Runtime/LuaVirtualMachine.cs:595]: Perhaps you forgot to remove the debugging output.

File: src/Lua/Runtime/LuaVirtualMachine.cs
Comment (AnnulusGames) [inline src/Lua/Runtime/LuaVirtualMachine.cs:1]: In order to maintain code consistency throughout this library, I would appreciate it if you could remove the `private` keyword.

Comment (Akeit0) [reply src/Lua/Runtime/LuaVirtualMachine.cs:1]: Do you want to erase the private modifier in LuaValueDictionary?

File: src/Lua/LuaThread.cs [hunk 2]
```diff
@@ -35,7 +36,14 @@ internal void PushCallStackFrame(CallStackFrame frame)
     {
         callStack.Push(frame);
     }
+    
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    internal void PushCallStackFrame(ref CallStackFrame frame)
```
[the remark this answers is not in the document]
Comment (Akeit0) [reply src/Lua/LuaThread.cs:41]: Can I delete PushCallStackFrame passing CallStackFrame by value?

File: src/Lua/Runtime/LuaVirtualMachine.cs
Comment (AnnulusGames) [reply src/Lua/Runtime/LuaVirtualMachine.cs:1]: As for LuaValueDictionary, it is a modification of the Dictionary implementation, so there is no need to change it.

File: src/Lua/LuaThread.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (AnnulusGames) [reply src/Lua/LuaThread.cs]: Yes, please.

**Comment by Akeit0** [comment]
I think I've finished revising them according to your comment.

**Comment by AnnulusGames** [comment]
I haven't looked into the implementation of LuaVirtualMachine in detail yet, so I'll merge it in as soon as I've checked it.

**Comment by AnnulusGames** [comment]
I have checked the VM implementation. It looks very good. The only thing that bothers me is that SetOperations is separated into a separate Operators class. Is there a reason for this?

Also, I would also be grateful if you could let me know if you have any references for this optimization.

**Comment by AnnulusGames** [comment]
There is a conflict due to the merge of #47, could I resolve this on my end?

**Comment by Akeit0** [comment]
Of course!

**Comment by Akeit0** [comment]
> I have checked the VM implementation. It looks very good. The only thing that bothers me is that SetOperations is separated into a separate Operators class. Is there a reason for this?
> 
> Also, I would also be grateful if you could let me know if you have any references for this optimization.
Considering that LuaVirtualMachine is a static class, and that using lambda expressions reduces readability and the visibility of exceptions, I use nested class.

**Comment by Akeit0** [comment]
> Also, I would also be grateful if you could let me know if you have any references for this optimization.

No references available. I looked at the generated code and found that it was less efficient, so I adjusted it to generate the optimal code

**Comment by AnnulusGames** [comment]
I've resolved the conflicts and formatted the code.

If possible, could you please add comments to each step of the code? This will help me and other contributors understand the code properly.

**Comment by Akeit0** [comment]
@AnnulusGames 
OK! Shall I leave the original async code in the comments?

**Comment by Akeit0** [comment]
Are these comments sufficient?

**Pull request closed without merging**